helpers/discordAuth.ts

```diff
@@ -0,0 +1,77 @@
+import { prisma } from '../prisma'
+import { URLSearchParams } from 'url'
+
+const client_id = process.env.DISCORD_KEY
+const client_secret = process.env.DISCORD_SECRET
+
+type DiscordUserInfo = {
+  discordUsername: string
+  discordAvatarUrl: string
+  discordRefreshToken: string
+}
+
+export const getTokenFromAuthCode = (code: string) => {
```

Comment:
Please add the return type for these JSONs. Don't wanna have to deal with `any` types specially with authentication.

https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-access-token-response